lib/index.js

```diff
@@ -13,38 +37,64 @@ module.exports = {
     const variant = config.variant;
     return {
       upload: async (file) => {
-        switch (file.mime.split('/')[0]) {
-          case 'video':
-            const uploadVideoResponse = await uploadVideo(file, accountId, apiKey);
-            const streamMediaId = uploadVideoResponse._response.headers['stream-media-id'];
-            file.url = `https://watch.videodelivery.net/${streamMediaId}`;
-            file.preview_url = `https://videodelivery.net/${streamMediaId}/thumbnails/thumbnail.jpg`;
-            file.provider_metadata = {
-              public_id: streamMediaId,
-              dash_url: `https://videodelivery.net/${streamMediaId}/manifest/video.mpd`,
-              hls_url: `https://videodelivery.net/${streamMediaId}/manifest/video.m3u8`,
-            };
-            return file
-          case 'image':
-            const uploadImageResponse = await uploadImage(file, accountId, apiKey);
-            const result = uploadImageResponse.data.result;
-            const filename = result.filename;
-            const split = filename.split('.');
-            const type = split.length > 0 ? split[split.length - 1] : '';
-            file.url = result.variants.find(v => v.includes('/' + variant));
-            file.provider_metadata = {
-              public_id: result.id,
-              resource_type: type,
-              variant_url: file.url.substr(0, file.url.lastIndexOf('/'))
-            };
-            return file
+        switch (file.mime.split("/")[0]) {
+          case "video":
+            try {
+              const streamMediaId = await uploadVideo(file, accountId, apiKey);
+              return handleVideoResponse(file, streamMediaId);
+            } catch (e) {
+              // strapi does not clear the temp file if an error is thrown
+              clearTempFile();
+              throw e;
+            }
+          case "image":
+            try {
```

Comment:
could we have wrapped the whole `case` block in this try catch?

that would also catch the `Error("Unhandled File Type")`

Reply:
you are right I will update it